shared/middleware/helpers/auth_helpers.rb

```diff
@@ -23,6 +23,17 @@ def admin?
   current_user && !!current_user[:admin]
 end
 
+# @param [String] permission - Name of the permission we're interested in
+# @returns [Boolean] true if the current user has the specified dashboard permission
+def has_permission?(permission)
+  return false unless current_user
+
+  @user_permissions ||= {}
+  return @user_permissions[permission] unless @user_permissions[permission].nil?
+  result = DASHBOARD_DB[:user_permissions].where(user_id: current_user_id, permission: permission).any?
```

Comment:
This caches each permission individually, causing repeated method invocations to possibly hit the DB multiple times. Is there any reason not to grab all of current_user's permissions at once (e.g., like https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/models/user.rb#L148-L160)?